src/porepy/numerics/ad/operators.py

```diff
@@ -649,11 +650,48 @@ def _identify_subtree_discretizations(self, discr: list) -> list:
 
     ### Operator parsing ---------------------------------------------------------------
 
+    def value(
+        self, system_manager: pp.ad.EquationSystem, state: Optional[np.ndarray] = None
+    ) -> numbers.Real | np.ndarray | sps.spmatrix:
+        return self._evaluate(system_manager, state=state, evaluate_jacobian=False)
+
+    def value_and_jacobian(
+        self, system_manager: pp.ad.EquationSystem, state: Optional[np.ndarray] = None
+    ) -> AdArray:
+        ad = self._evaluate(system_manager, state=state, evaluate_jacobian=True)
+
+        # Casting the result to AdArray or raising an error.
+        if isinstance(ad, numbers.Real):
+            # AdArray requires 1D numpy array as value, not a scalar.
+            ad = np.array(ad).reshape(1)
```

Comment:
Is this equivalent? If so, I prefer the below suggestion, but I see that is a matter of preference.
```suggestion
            ad = np.array([ad])
```